src/conways/conways_game_view.rs

```diff
@@ -0,0 +1,55 @@
+use std::{thread::sleep, time::Duration};
+
+use macroquad::{
+    color::{BLACK, WHITE},
+    shapes::draw_rectangle,
+    window::{clear_background, next_frame, screen_height, screen_width},
+};
+
+use super::{
+    conways_game::ConwaysGame,
+    conways_game_constants::{CELLS_HEIGHT, CELLS_WIDTH, CONSTANT_WAIT, VIEW_SCALE_FACTOR},
+};
+
+pub struct ConwaysGameView {
+    conways_game: ConwaysGame,
+    is_drawing: bool,
+}
+
+impl ConwaysGameView {
+    pub fn new(conways_game: ConwaysGame) -> Self {
+        Self {
+            conways_game,
+            is_drawing: true,
+        }
+    }
+
+    pub async fn start_drawing(&mut self) {
+        while self.is_drawing {
+            clear_background(WHITE);
+
+            self.conways_game.next_generation();
+
+            self.conways_game.cells_do(|cell| {
+                let width = CELLS_WIDTH;
+                let height = CELLS_HEIGHT;
+                let scale_factor = VIEW_SCALE_FACTOR;
+                let x_position = Self::get_x_position(cell.x_position, scale_factor);
+                let y_position = Self::get_y_position(cell.y_position, scale_factor);
+                draw_rectangle(x_position, y_position, width, height, BLACK);
+            });
+
+            sleep(Duration::from_secs_f32(CONSTANT_WAIT));
```

Comment:
Instead of using a `sleep`, which blocks the execution thread, I would consider using a timeout. At low [`CONSTANT_WAIT`](https://github.com/belfortep/conways-game-in-rust/blob/79e214451dede6d50798231592ea29896499efe2/src/conways/conways_game_constants.rs#L5) it works fine, but I tried increasing the constant to slow the simulation down, and then interacting with the window controls (resizing, fullscreen, closing), which was buggy (I'm guessing, because of the execution blocking at the `sleep`).

Reply:
Okay, just searched and there are some function for controlling the time in macroquad, I'm trying with those

Reply:
Nice! I was using `Instant` haha. If there is time utilities I'll feel dumb 🤣